presto-main/src/test/java/com/facebook/presto/execution/TestNodeScheduler.java

```diff
@@ -424,6 +425,28 @@ public void testSplitCount()
         assertEquals(nodeTaskMap.getPartitionedSplitsOnNode(chosenNode), 0);
     }
 
+    @Test
+    public void testMaxTasksPerStage()
+    {
+        TestingTransactionHandle transactionHandle = TestingTransactionHandle.create();
+
+        ImmutableList.Builder<Split> initialSplits = ImmutableList.builder();
+        for (int i = 0; i < 10; i++) {
+            initialSplits.add(new Split(CONNECTOR_ID, transactionHandle, new TestSplitRemote()));
+        }
+
+        Set<Split> splits = new HashSet<>();
+        for (int i = 0; i < 5; i++) {
+            splits.add(new Split(CONNECTOR_ID, transactionHandle, new TestSplitRemote()));
+        }
+        Multimap<InternalNode, Split> assignments = nodeSelector.computeAssignments(splits, ImmutableList.copyOf(taskMap.values()), 2).getAssignments();
+        assertEquals(assignments.keySet().size(), 2);
+        Multimap<InternalNode, Split> tmp = assignments;
+        assignments = nodeSelector.computeAssignments(splits, ImmutableList.copyOf(taskMap.values()), 2).getAssignments();
+        // second call should return same nodes
```

Comment:
What behavior are we testing here where the same set of nodes is returned each time? When would we want to get the list of nodes multiple times deterministically?

Reply:
Actually yes, we want to test that if `computeAssignments` is called multiple times, no more than the nodes limit is assigned.